libs/dev-dock/backend/src/backend.ts

```
@@ -0,0 +1,176 @@
+    assert(parseResult.isOk());
+    const electionInfo: DevDockElectionInfo = {
+      path: input.path,
+      title: parseResult.ok().election.title,
+    };
+
+    writeDevDockFileContents({
+      electionInfo,
+    });
+  },
+
+  getElection(): Optional<DevDockElectionInfo> {
+    return readDevDockFileContents().electionInfo;
+  },
+
+  getCardStatus(): CardStatus {
+    return readFromCardMockFile().cardStatus;
+  },
+
+  async insertCard(input: { role: DevDockUserRole }): Promise<void> {
+    const electionDefinitionPath = readDevDockFileContents().electionInfo?.path;
+    if (!electionDefinitionPath) {
+      return;
+    }
+
+    await execFile(MOCK_CARD_SCRIPT_PATH, [
+      '--card-type',
+      input.role.replace('_', '-'),
+      '--electionDefinition',
+      electionPathToAbsolute(electionDefinitionPath),
+    ]);
+  },
+
+  async removeCard(): Promise<void> {
+    await execFile(MOCK_CARD_SCRIPT_PATH, ['--card-type', 'no-card']);
+  },
+
+  getUsbDriveStatus(): DevDockUsbDriveStatus {
+    return getUsbDriveStatus();
+  },
+
+  async insertUsbDrive(): Promise<void> {
+    if (getUsbDriveStatus() === 'removed') {
+      await insertUsbDrive();
+    }
+  },
+
+  async removeUsbDrive(): Promise<void> {
+    if (getUsbDriveStatus() === 'inserted') {
+      await removeUsbDrive();
+    }
+  },
+
+  async clearUsbDrive(): Promise<void> {
+    if (getUsbDriveStatus() === 'inserted') {
+      await removeUsbDrive();
+      await clearUsbDrive();
+      await insertUsbDrive();
+    } else {
+      await clearUsbDrive();
+    }
+  },
+});
+
+/**
+ * A type to be used by the frontend to create a Grout API client
+ */
+export type Api = typeof api;
+
+/**
+ * Mounts the Dev Dock API endpoints at /dock.
+ */
+export function useDevDockRouter(
+  app: Express.Application,
+  express: typeof Express
+): void {
+  if (!isFeatureFlagEnabled(BooleanEnvironmentVariableName.ENABLE_DEV_DOCK)) {
+    return;
+  }
```

Maybe we could throw an error if `ENABLE_DEV_DOCK` is on but `MOCK_SMART_CARDS` is not?

---

It might be useful to allow the other dev dock tools even while using real smartcards. I'd rather just disable the card interface